src/bootman/config.c

```diff
@@ -58,44 +58,92 @@ bool boot_manager_set_timeout_value(BootManager *self, int timeout)
                 return false;
         }
 
-        if (fprintf(fp, "%d\n", timeout) < 0) {
+        if (fprintf(fp, "%s\n", contents) < 0) {
                 LOG_FATAL("Unable to set new timeout: %s", strerror(errno));
                 return false;
         }
         return true;
 }
 
-int boot_manager_get_timeout_value(BootManager *self)
+bool boot_manager_set_timeout_value(BootManager *self, int timeout)
+{
+        if (timeout <= 0) {
+            return _write_sysconf_file(self, "timeout", NULL);
+        }
+
+        autofree(char) *timeout_s = string_printf("%d", timeout);
+
+        return _write_sysconf_file(self, "timeout", timeout_s);
+}
+
+bool boot_manager_set_console_mode(BootManager *self, const char *mode)
+{
+        if (mode == NULL) {
+            return _write_sysconf_file(self, "console_mode", NULL);
+        }
+
+        return _write_sysconf_file(self, "console_mode", mode);
```

Comment:
The NULL check seems a bit pointless here as you still pass NULL. Perhaps it's better to ensure all the `self` vtable functions do an assert against `self` being valid per the older APIS?